src/keymap-extensions.coffee

```diff
@@ -20,12 +20,37 @@ KeymapManager::getUserKeymapPath = ->
 
 KeymapManager::loadUserKeymap = ->
   userKeymapPath = @getUserKeymapPath()
-  if fs.isFileSync(userKeymapPath)
+  return unless fs.isFileSync(userKeymapPath)
+
+  try
     @loadKeymap(userKeymapPath, watch: true, suppressErrors: true)
+  catch error
+    if error.message.indexOf('Unable to watch path') > -1
+      message = """
+        Unable to watch path: `#{path.basename(userKeymapPath)}`. Make sure you
+        have permission to read `#{userKeymapPath}`.
+
+        On linux there are currently problems with watch sizes. See
+        [this document][watches] for more info.
+        [watches]:https://github.com/atom/atom/blob/master/docs/build-instructions/linux.md#typeerror-unable-to-watch-path
+      """
+      atom.notifications.addError(message, {dismissable: true})
+    else
+      detail = error.path
+      stack = error.stack
+      atom.notifications.addFatalError(error.message, {detail, stack, dismissable: true})
 
 KeymapManager::subscribeToFileReadFailure = ->
-  this.onDidFailToReadFile (error) ->
-    atom.notifications.addError('Failed to load keymap.cson', {detail: error.stack, dismissable: true})
+  this.onDidFailToReadFile (error) =>
```

Comment:
`@` instead of `this`? I see it was using `this` before, but might be nice to change it now since it is being edited.